extensions/ql-vscode/src/skeleton-query-wizard.ts

```diff
@@ -231,17 +231,29 @@ export class SkeletonQueryWizard {
   }
 
   private async selectOrDownloadDatabase() {
+    if (this.language === undefined) {
+      throw new Error("Language is undefined");
+    }
+
     if (this.qlPackStoragePath === undefined) {
       throw new Error("QL Pack storage path is undefined");
     }
 
-    const existingDatabaseItem = await this.findExistingDatabaseItem();
+    const existingDatabaseItem = await this.findExistingDatabaseItem(
+      this.language,
+      this.databaseManager.databaseItems,
+    );
 
     if (existingDatabaseItem) {
       await this.databaseManager.setCurrentDatabaseItem(existingDatabaseItem);
     } else {
       await this.downloadDatabase();
```

Comment:
This isn't directly related to this PR but I've noticed that we could "easily" (famous last words) make `downloadDatabase` return the newly-created database item. It seems that `downloadGitHubDatabase` already returns it, so we'd just need to return it from `downloadDatabase`. That would remove the need to call `findExistingDatabaseItem` afterwards and potentially remove a chance of race conditions.

Reply:
I also just noticed when reviewing https://github.com/github/vscode-codeql/pull/2328 that `downloadDatabase` should theoretically already set the new database as the currently selected item. It calls through to [`databaseArchiveFetcher`](https://github.com/github/vscode-codeql/blob/4170e7f7a7026b3ebcc75b7509dd55f317293a4d/extensions/ql-vscode/src/databaseFetcher.ts#L273) which calls [`setCurrentDatabaseItem`](https://github.com/github/vscode-codeql/blob/4170e7f7a7026b3ebcc75b7509dd55f317293a4d/extensions/ql-vscode/src/databaseFetcher.ts#L326).

Would be good to test if this is working. If it isn't could it be the same single-workspace issue, so once that is fixed this will also work?

Reply:
> If it isn't could it be the same single-workspace issue, so once that is fixed this will also work?

Yes, that's currently what's happening. Because we decided to select the database when we open it, we no longer need to repeat the step here.